spec.emu

```diff
@@ -111,6 +111,32 @@ contributors: Ujjwal Sharma, Younies Mahmoud
       </table>
     </emu-table>
 
+    <emu-clause id="sec-todurationrecord" type="abstract operation">
+      <h1>
+        ToDurationRecord (
+          _input_: an Object
+        ): a Record with the fields listed in the Value Slot column of <emu-xref href="#table-duration-components"></emu-xref>
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It converts a given object that represents a Duration into a Record.</dd>
+      </dl>
+
+      <emu-alg>
+        1. Let _result_ be a new Record.
+        1. For each row in <emu-xref href="#table-duration-components"></emu-xref>, except the header row, in table order, do
+          1. Let _valueSlot_ be the Value Slot value.
+          1. Let _unit_ be the Unit value.
+          1. Let _value_ be ? Get(_input_, _unit_).
+          1. If _value_ is not *undefined*, then
+            1. Set _value_ to ? ToIntegerWithoutRounding(_value_).
+          1. Else,
+            1. Set _value_ to `0`.
+          1. Set the field of _result_ whose name is _valueSlot_ to _value_.
+        1. Return _result_.
```

Comment:
For consistency with Temporal, I think this operation should throw if none of the recognized units are present, e.g. _input_ is `{}` or `{ megaseconds: 3 }`